src/novmm/machine/virtio.go

```diff
@@ -258,6 +262,8 @@ func (vchannel *VirtioChannel) consumeOne() (bool, error) {
 }
 
 func (vchannel *VirtioChannel) consumeOutstanding() error {
+	vchannel.outstandingLock.Lock()
```

Comment:
The outstandingLock will be acquired by processOne below, no? Isn't this a deadlock condition?